src/Compilers/CSharp/Portable/Symbols/SubstitutedNamedTypeSymbol.cs

```diff
@@ -438,5 +438,20 @@ internal override IEnumerable<CSharpAttributeData> GetCustomAttributesToEmit(PEM
         internal sealed override bool IsRecord => _underlyingType.IsRecord;
         internal sealed override bool IsRecordStruct => _underlyingType.IsRecordStruct;
         internal sealed override bool HasPossibleWellKnownCloneMethod() => _underlyingType.HasPossibleWellKnownCloneMethod();
+
+#nullable enable
+        internal sealed override bool IsExtension => _underlyingType.IsExtension;
+
+        protected sealed override TypeSymbol? ExtensionUnderlyingTypeNoUseSiteDiagnosticsCore
+            => _unbound ? null : Map.SubstituteType(OriginalDefinition.ExtensionUnderlyingTypeNoUseSiteDiagnostics).Type;
+
+        protected sealed override ImmutableArray<NamedTypeSymbol> BaseExtensionsNoUseSiteDiagnosticsCore
+            => _unbound ? ImmutableArray<NamedTypeSymbol>.Empty : Map.SubstituteNamedTypes(OriginalDefinition.BaseExtensionsNoUseSiteDiagnostics);
+
+        internal sealed override TypeSymbol? GetDeclaredExtensionUnderlyingType()
+            => IsExtension ? throw new InvalidOperationException("PROTOTYPE") : throw ExceptionUtilities.Unreachable();
```

Comment:
>IsExtension [](http://example.com/codeflow?start=15&length=11)

It feels wrong to condition behavior on this value. We didn't do this for base type and interfaces.